test/powershell/engine/Remoting/RemoteSession.Basic.Tests.ps1

```diff
@@ -10,47 +9,266 @@ Describe "New-PSSession basic test" -Tag @("CI") {
+        }
+    }
+
+    It 'Can connect to default endpoint' {
+        $session = New-RemoteSession -ConfigurationName $endPoint
+        ValidateSessionInfo -session $session -state 'Opened'
+        $session | Remove-PSSession -ErrorAction SilentlyContinue
+    }
+
+    It 'Can execute command in a disconnected session' {
+        $session = Invoke-RemoteCommand -InDisconnectedSession -ComputerName 'localhost' -ScriptBlock { 1+1 } -ConfigurationName $endPoint
+
+        ValidateSessionInfo -session $session -state 'Disconnected'
+
+        $result = Receive-PSSession -Session $session
+        $result | Should Be 2
+        $result.PSComputerName | Should BeExactly 'localhost'
+
+        $session | Remove-PSSession -ErrorAction SilentlyContinue
+    }
+
+    It 'Can disconnect and connect to PSSession' {
+        $session = New-RemoteSession -ConfigurationName $endPoint
+
+        ValidateSessionInfo -session $session -state 'Opened'
+        Disconnect-PSSession -Session $session
+
+        ValidateSessionInfo -session $session -state 'Disconnected'
+        Connect-PSSession -Session $session
+
+        $result = Invoke-Command -Session $session -ScriptBlock { 1 + 1 }
+        $result | Should Be 2
+        $result.PSComputerName | Should BeExactly 'localhost'
+
+        $session | Remove-PSSession -ErrorAction SilentlyContinue
+    }
+
+    It 'Can export and import PSSession' {
+        $name = Get-Random
+        $commandName = 'Add-Number'
+
+        Invoke-Command -Session $openSession -ScriptBlock { function Add-Number ($number1, $number2) { $number1 + $number2 } }
+
+        $null = Export-PSSession -OutputModule $name -Force -Session $openSession -CommandName $commandName
+        $imported = Import-PSSession -Module $name -Session $openSession
+
+        $imported.ExportedCommands.Keys | Should BeExactly $commandName
+        Remove-Item "function:\$commandName" -ErrorAction SilentlyContinue -Force
+    }
+
+    It "<title>" -TestCases $ParameterError {
+        param($parameters, $expectedError)
+
+        { Invoke-Command @parameters } | ShouldBeErrorId $expectedError
+    }
+
+    It 'Can execute command if one of the sessions is available' {
+        try
+        {
+            $result = Invoke-Command -Session $openSession,$disconnectedSession,$closedSession -ScriptBlock { 1+1 } -ErrorAction SilentlyContinue
+        }
+        catch
+        {
+            if($_.FullyQualifiedErrorId -ne 'InvokeCommandCommandInvalidSessionState,Microsoft.PowerShell.Commands.InvokeCommandCommand')
+            {
+                # We expect the error from $disconnectedSession and $closedSession. Hence, throw otherwise.
+                throw $_
+            }
+        }
+
+        $result.Count | Should Be 1
+        $result | Should Be 2
+    }
+
+    It 'Can execute command without creating new scope' {
+        $result = Invoke-Command -NoNewScope -ScriptBlock { 1 + 1 }
```

Comment:
This test doesn't validate that the scriptblock was run in the correct scope. It just validates that the scriptblock was run in a scope. It provides code coverage, but isn't really useful in terms of validating functionality as-is.

Reply:
Fixed. I set a variable in the script block and check the value after.